### DISCUSSION TITLE
local embedding in python without ollama running

### DISCUSSION THREAD

**Discussion by swaipdev** [mem0ai/mem0#2015] [Knowledge Base]
Hi, for an offline app, I am trying to embed locally with just python (I don't want my user to have to install Ollama to do this). 
Is this possible in mem0 ?
Thanks in advance

**Comment by swaipdev** [comment]
This would work (I imagine) by creating a new local embedding class in mem0/embeddings, that creates embeddings locally. 
Then I would call it like this:

config = {
    "embedder": {
        "provider": "local",
        "config": {
            "model": "sentence-transformers/all-MiniLM-L6-v2",  # Local model name
            "embedding_dims": 384  # Set dimensions according to your model
        }
    }
}

Does this approach make any sense ?

**Reply by Dev-Khant** [reply]
Hi @swaipdev To run embedding locally you will have to install Ollama.
